.github/workflows/test.yml

```diff
@@ -147,15 +165,27 @@ jobs:
           /usr/bin/sed -i.bak \
             "s/# programs.fish.enable = true;/nix.settings.access-tokens = [ \"github.com=\${{ secrets.GITHUB_TOKEN }}\" ]; environment.etc.\"nix\/nix.conf\".knownSha256Hashes = [ \"$nixConfHash\" ];/" \
             flake.nix
+          /usr/bin/sed -i.bak \
+            's/nixpkgs.hostPlatform = "aarch64-darwin";/nixpkgs.hostPlatform = "'$(nix eval --expr builtins.currentSystem --impure --raw)'";/' \
```

Comment:
I'm sure it's fine because we're using other nix3 commands, but I don't see where we're setting experimental features to support that, so this gives me a touch of the heebie-jeebies

Reply:
The `cachix/install-nix-action` adds `nix-command flakes` to `experimental-features` by default

I've updated to remove unnecessary `--extra-experimental-features`, the only place it is actually necessary is after we switch into a non-flake nix-darwin configuration as it doesn't set `nix-command`